### ISSUE TITLE
Severe Performance Hit When MultiSampleCount enabled

### ISSUE THREAD

**Issue by captkirk88** [MyraUI/Myra#32]
Try running with 
```csharp
GraphicsDeviceManager.PreferMultiSampling = true;
PresentationParameters.MultiSampleCount = 16 or 8 or 4 or 2; // different variations of performance hit
```
You will see it run to a crawl or outright stop responding.
A better way to control the underlying SpriteBatch implementation would help but I don't know if it would fix the way Myra draws internally.

**Comment by captkirk88** [comment]
Not really related to Anti-aliasing unless you are changing the RasterizerState but it still impacts it nevertheless.

**Comment by captkirk88** [comment]
Also, it should be noted that I have 3D assets behind my gui.
![image](https://user-images.githubusercontent.com/6296457/44072612-185bb0e6-9f4d-11e8-99b2-0af7dbaa028e.png)

**Comment by rds1983** [comment]
Did I understand correctly that the issue could be fixed by allowing a user to control what parameters are passed to Desktop's SpriteBatch.Begin?

**Comment by captkirk88** [comment]
I think opening up access to the batch would allow the user to better tune it to their draw strategy.

If it is as easy as opening up access to the RenderContext with a property in Desktop and allowing that to be set, that would really help.  Especially since RenderContext.Batch can be set.

**Comment by captkirk88** [comment]
Another benefit of that change would allow the user to set the projection matrix themselves, on their SpriteBatch before the call to Desktop.Render, to create that sweet 3D interface (not what I was doing but it would be possible).  Especially since you opened up the MouseStateGetter property.  Well to do that you would have to separate the UpdateLayout() UpdateInput() in Desktop.Render into a Desktop.Update method.  I do not know what other changes you might need to do to allow custom SpriteBatch implementation.  Would Desktop.Bounds be affected?

**Comment by rds1983** [comment]
Ok, I'll expose SpriteBatch in the next version (0.6.7)
Again thanks for the idea!

**Comment by rds1983** [comment]
Unfortunately I can't break Desktop.Render as you proposed and allow user to use his own SpriteBatch. As SpriteBatch.Begin could be called multiple times during UI rendering (when Widget.ClipToBounds is set to true then SpriteBatch.End/SpriteBatch.Begin is called, because ScissorRectangle was changed).

However I've added property SpriteBatchBeginParams to the Desktop which allows a user to customize what parameters passed to the SpriteBatch.Begin. The new property is available in the latest version (0.6.7).
